lib/actions/appleMusic.js

```diff
@@ -155,15 +145,16 @@ function appleMusic(player, values) {
                 return player.coordinator.clearQueue()
                   .then(() => player.coordinator.addURIToQueue(queueTracks[0].uri, queueTracks[0].metadata, true, 1))
                   .then(() => {
-                      if (radioOn(player)) {
-                        return promise.then(() => player.coordinator.setAVTransport(queueURI,''));
+                      if (isRadioOrLineIn(player.coordinator.state.currentTrack.uri)) {
+                        player.coordinator.setAVTransport(queueURI,''));
                       }
+                      return Promise.resolve();
                     })
                   .then(() => player.coordinator.play())
                   .then(() => {
                     return queueTracks.reduce(function(promise, track, index){
                       if (index > 0) {
-                        return promise.then(() => player.coordinator.addURIToQueue(track.uri, track.metadata, true, index + 1));
+                        player.coordinator.addURIToQueue(track.uri, track.metadata, true, index + 1);
                       } else {
                         return promise.then(() => {});
                       }   
```

Comment:
I get the confusion. 

1. This is not actually related to Promises, but rather arrow functions which are similar to arrow functions in C# (and perhaps Java as well). https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions. Arrow functions without body (basically single-line without the {}) are direct returning. 

2. First argument to the reduce delegate/callback is always the value returned from the previous run. You don't have an initial value (which is the second argument to reduce(), which means that it will be the first value in the array instead (which isn't a promise). So this won't work:

```
[1,2,3].reduce((promise, value) => promise.then())
```
but this will:
```
[1,2,3].reduce((promise, value) => promise.then(), Promise.resolve())
```

In the first example, promise will be the literal value 1 at first run, and value will be 2. 
In the second example, it will be a simple promise, because of the second argument to reduce. value will 1 at first run.